webdriver-spec.html

```diff
@@ -1435,288 +1435,532 @@
-    else call <a>set the proxy</a> passing in <var>proxy</var>.
-
-   <li><p>Let <var>page load strategy</var> be the result of <a>getting a property</a>
-    named <code>pageLoadStrategy</code> from <var>unprocessed capability</var>.
-    If <var>page load strategy</var> is <a>undefined</a>,
-    then set the entry <code>pageLoadStrategy</code>
-    in <var>server capabilities</var> to <a>normal</a>.
-  </ol>
+  <li><p>If <var>merged capabilities</var> is an <a>error</a>,
+  return the <var>merged capabilities</var> error.
 
- <li><p>If the length of <var>unmet capabilities</var> is not equal to 0,
-  return <a>session not created</a> with data <var>unmet capabilities</var>.
+  <li><p>Set <var>matched capabilities</var> to the result of
+  <a>matching capabilities</a> with <var>merged capabilities</var> as an
+  argument.
 
- <li><p>Return <var>server capabilities</var>.
+  <li><p>Increment <var>k</var> by 1.
+  </ol>
+
+<li><p>Return <var>matched capabilities</var>.
 </ol>
 
-<p>To <dfn>set the proxy</dfn> from a JSON Object <var>proxy</var>:
+<p>When <dfn>merging capabilities</dfn> with JSON Object arguments
+  <var>primary</var> and <var>secondary</var>, an endpoint node
+  <em>MUST</em> run the following steps:
 
 <ol>
- <li><p>Let <var>proxy type</var> be the result of
-  <a>getting a property</a> named <code>proxyType</code> from <var>proxy</var>.
+<li><p>Let <var>result</var> be a JSON Object with the same entries as
+<var>primary</var>.
 
- <li><p>Switch on <var>proxy type</var>:
+<li><p>If <var>secondary</var> is <a>null</a> or <a>undefined</a>, return
+<var>result</var>.
 
-  <dl class=switch>
-   <dt>"<code>pac</code>"
-   <dd>Let <var>proxy autoconfiguration url</var> be the result of <a>getting a
-    property</a> named <code>proxyAutoconfigUrl</code> from <var>proxy</var>.
+<li><p>Let <var>k</var> be 0.
 
-    <p>If the implementation supports <a>proxy autoconfiguration</a>
-    and <var>proxy autoconfiguration url</var> is defined, set the
-    implementation’s autoproxy configuration URL to <var>proxy autoconfiguration
-    url</var>.
+<li><p>Let <var>num entries</var> be the number of entries in
+<var>secondary</var>.
 
-    <p>Otherwise return an <a>error</a>
-     with <a>error code</a> <a>invalid argument</a>.
+<li><p>While <var>k</var> &lt; <var>num entries</var>:
 
-   <dt>"<code>noproxy</code>"
-   <dd><p>Set the proxy to No Proxy using implementation defined steps.
-    If this can not be set during this process return <a>error</a>.
+  <ol>
+  <li><p>Let <var>entry</var> be the entry in <var>secondary</var> at index
+  <var>k</var>.
 
-   <dt>"<code>autodetect</code>"
-   <dd><p>Set the proxy to Auto-Detect proxy setting from the network
-    using implementation defined steps.
-    If this can not be set during this process return <a>error</a>.
+  <li><p>If the name of <var>entry</var> is also the name of an entry in
+  <var>result</var>,
+  return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
```

Comment:
should we process all the entries and keep track of errors and if `errors.length > 0` then raise the errors?

Doing that might catch multiple issues rather than people fix one thing retry, fix the next thing, retry, fix the next thing and so on.